afs-ws/afs-ws-client-utils/src/main/java/com/powsybl/afs/ws/client/utils/ClientUtils.java

```diff
@@ -114,16 +113,11 @@ public static void checkOk(Response response) {
 
     public static <T> Optional<T> readOptionalEntityIfOk(Response response, Class<T> entityType) {
         Response.Status status = Response.Status.fromStatusCode(response.getStatus());
-        if (status == Response.Status.OK) {
-            return Optional.of(readEntityAndLog(response, entityType));
-        } else if (status == Response.Status.NO_CONTENT) {
+        if (status == Response.Status.NO_CONTENT || status == Response.Status.NOT_FOUND) {
```

Comment:
Maybe add a comment that the NO_CONTENT is for backwards compatibility with afs-server < version XX